sys/include/delay.h

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2023 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @defgroup    sys_delay   Convenience functions to delay program execution
+ * @ingroup     sys
+ * @brief       Convenience functions to delay program execution
+ *
+ * # Motivation
+ *
+ * @ref sys_ztimer already provides @ref ztimer_sleep to allow delaying the
+ * execution of the running thread. While @ref ztimer_sleep allows via the
+ * selection of the clock to select the best fitting trade-off between
+ * resolution, accuracy (clock drift), power consumption and so on, it is
+ * difficult to write code with flexible requirements that just picks whatever
+ * clock is available.
+ *
+ * In addition, this also adds @ref delay_us_spinning as a wrapper for
+ * @ref ztimer_spin which was found to be too slow on AVR to implement some
+ * bit-banging drivers. Instead, this uses a custom implementation for AVR
+ * that is more accurate and easily allows other platforms to also CPU cycle
+ * counting implementations.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Convenience functions to delay program execution
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@posteo.net>
+ */
+
+#ifndef DELAY_H
+#define DELAY_H
+
+#include <assert.h>
+#include <stdint.h>
+
+#include "compiler_hints.h"
+#include "delay_arch.h"
+#include "time_units.h"
+#include "ztimer.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Delay program execution for roughly the given amount of
+ *          milliseconds
+ *
+ * @param[in]   msecs       Duration to delay execution in milliseconds
+ *
+ * @note        This function gives no guarantees regarding accuracy or
+ *              power saving. Use @ref ztimer_sleep instead when you care.
+ */
+static inline void delay_ms(uint32_t msecs)
+{
+    if (IS_USED(MODULE_ZTIMER_MSEC)) {
```

Comment:
```suggestion
    if (IS_USED(MODULE_ZTIMER_MSEC)  &&  msec > 10) {
```
jitter < 10%

Reply:
This would trade ROM for increase accuracy. But the whole point of this API is to trade in accuracy to safe ROM / dependencies.

Reply:
- ztimer_msec is not able to sleep for 1 ms ( a value of 1 is 0 usec to 2000 usec).
- there will not be any ROM usage for compiletime constant delays i would such a functions be used for (static inline)